samcli/local/docker/container.py

```diff
@@ -200,7 +199,7 @@ def wait_for_logs(self, stdout=None, stderr=None):
         real_container = self.docker_client.containers.get(self.id)
 
         # Fetch both stdout and stderr streams from Docker as a single iterator.
-        logs_itr = attach(self.docker_client, container=real_container, stdout=True, stderr=True, logs=True)
+        logs_itr = real_container.attach(stream=True, logs=False, demux=True)
```

Comment:
Setting `logs=False` changes the previous behaviour.  Is this intentional?

Reply:
@thomas-fossati What behavior. I have asked a similar question on #1566. It is not clear to me what you are @setrofim are referring to. Please be more specific.

Reply:
Apologies for the extreme terseness of my previous comment.

If the lambda business logics completes before SAM attaches to the container, the lambda's output will only be available through the logs.  So if you don't ask the logs to be forwarded on attach, you might end up seeing no output, including the bits that SAM cares about (ie., `{ "status": ... }`) and get stuck.

Reply:
@setrofim had a small back and forth here: https://github.com/awslabs/aws-sam-cli/pull/1566#issuecomment-557115763 I think I understand the reasoning more.

It was intentionally `False` because I took this from the 'warm container' PR in which we do not want all the logs from the container. Since we are not in that mode yet, making this `True` makes sense to me. We will need to filter logs by invocation for the warm container case but we can tackle all that at once.

Reply:
Sounds like a good plan to me, thanks!